src/main/java/appeng/recipes/factories/recipes/PartRecipeFactory.java

```diff
@@ -72,6 +72,48 @@ else if( type.contains( "shapeless" ) )
 		}
 	}
 
+	public static ItemStack getResult( JsonObject json, JsonContext context )
+	{
+		JsonObject resultObject = JsonUtils.getJsonObject( json, "result" );
+
+		if( resultObject.has( "part" ) )
+		{
+			return getPart( resultObject );
+		}
+		else if( resultObject.has( "item" ) )
+		{
+			return CraftingHelper.getItemStack( resultObject, context );
+		}
+		else
+		{
+			throw new JsonSyntaxException( "Result has no 'part' or 'item' property." );
+		}
+	}
+
+	private static ItemStack getPart( JsonObject resultObject )
+	{
+		String ingredient = JsonUtils.getString( resultObject, "part" );
+		Object result = (Object) Api.INSTANCE.registries().recipes().resolveItem( AppEng.MOD_ID, ingredient );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Object". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)